infra/RepoManager.py

```diff
@@ -0,0 +1,220 @@
+      raise RepoManagerException(
+          'Failed cloning repo %s, with error %s)' % (self.repo_url, err))
+    if not self._is_git_repo():
+      raise RepoManagerException('%s is not a git repo' % self.repo_url)
+
+  def _run_command(self, command, location='.'):
+    """ Runs a shell command in the specified directory location.
+
+    Args:
+      command: The command as a list to be run
+
+    Returns:
+      The stdout of the command, the stderr of the command
+    """
+    cur_dir = os.getcwd()
+    os.chdir(location)
+    process = subprocess.Popen(command, stdout=subprocess.PIPE)
+    out, err = process.communicate()
+    os.chdir(cur_dir)
+    if err is not None:
+      err = err.decode('ascii')
+      print('Error %s running command %s' % (err, command))
+    if out is not None:
+      out = out.decode('ascii')
+    return out, err
+
+  def _is_git_repo(self):
+    """Test if the current repo dir is a git repo or not.
+
+    Returns:
+      True if the current repo_dir is a valid git repo
+    """
+    git_path = os.path.join(self.repo_dir, '.git')
+    return os.path.isdir(git_path)
+
+  def _commit_exists(self, commit):
+    """ Checks to see if a commit exists in the project repo.
+
+    Args:
+      commit: The commit SHA you are checking for
+      project_name: The name of the project you are checking
+
+    Returns:
+      True if the commit exits in the project
+    """
+
+    # Handle the default case
+    if commit.strip(' ') == '':
+      return False
+
+    out, _ = self._run_command(['git', 'branch', '--contains', commit],
+                               self.repo_dir)
+    if ('error: no such commit' in out) or (
+        'error: malformed object name' in out) or (out == ''):
+      return False
+    return True
+
+  def get_current_commit(self):
+    """Gets the current commit SHA of the repo.
+
+    Returns:
+      The current active commit SHA
+    """
+    out, err = self._run_command(['git', 'rev-parse', 'HEAD'], self.repo_dir)
+    if err is not None:
+      return 1
+    return out.strip('\n')
+
+  def get_commit_list(self, old_commit, new_commit):
+    """Gets the list of commits(inclusive) between the old and new commits.
+
+    Args:
+      old_commit: The oldest commit to be in the list
+      new_commit: The newest commit to be in the list
+
+    Returns:
+      The list of commit SHAs from newest to oldest
+
+    Raises:
+      RepoManagerException when commits dont exist
+    """
+
+    if not self._commit_exists(old_commit):
+      raise RepoManagerException(
+          'The old commit %s does not exist' % old_commit)
+    if not self._commit_exists(new_commit):
+      raise RepoManagerException(
+          'The new commit %s does not exist' % new_commit)
+    if old_commit == new_commit:
+      return [old_commit]
+    out, err = self._run_command(
+        ['git', 'rev-list', old_commit + '..' + new_commit], self.repo_dir)
+    result = out.split('\n')
+    result = [i for i in result if i]
```

Comment:
why do we need `if i` part? are there any empty lines? can we just strip them before doing `out.split('\n')`?

Reply:
If there are two '\n's back to back, the split function will return '' as an element in the list.